database/mem_database_test.go

```diff
@@ -0,0 +1,49 @@
+package database
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestMemDatabase_Upsert(t *testing.T) {
```

Comment:
what's the point on adding this? there's already a list of tests running on different implementations, including the mem database.

Reply:
```
func (mb *MemDatabase) Upsert(namespace string, key string, value []byte) *DbError {
	mb.mu.Lock()
	defer mb.mu.Unlock()
	ns, ok := mb.namespaces[namespace]
	if !ok {
		ns = newNamespace()
		mb.namespaces[namespace] = ns
	}
	ns.data[key] = value
	return nil
}
```

I was not sure if the upsert function worked as expected, so I added this test. I thought `mb.namespaces[namespace]=ns` was necessary, but it turned out not.